x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/InferenceHelpers.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.inference.trainedmodel;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.xpack.core.ml.inference.results.ClassificationInferenceResults;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public final class InferenceHelpers {
+
+    private InferenceHelpers() { }
+
+    public static List<ClassificationInferenceResults.TopClassEntry> topClasses(List<Double> probabilities,
+                                                                                List<String> classificationLabels,
+                                                                                int numToInclude) {
+        if (numToInclude == 0) {
+            return Collections.emptyList();
+        }
+        int[] sortedIndices = IntStream.range(0, probabilities.size())
+            .boxed()
+            .sorted(Comparator.comparing(probabilities::get).reversed())
+            .mapToInt(i -> i)
+            .toArray();
+
+        List<String> labels = classificationLabels == null ?
+            // If we don't have the labels we should return the top classification values anyways, they will just be numeric
+            IntStream.range(0, probabilities.size()).boxed().map(String::valueOf).collect(Collectors.toList()) :
+            classificationLabels;
+
+        if (probabilities.size() != labels.size()) {
+            throw ExceptionsHelper
+                .badRequestException(
+                    "model returned classification probabilities of size [{}] which is not equal to classification labels size [{}]",
+                    probabilities.size(),
+                    classificationLabels);
+        }
+
+        int count = numToInclude < 0 ? probabilities.size() : numToInclude;
+        List<ClassificationInferenceResults.TopClassEntry> topClassEntries = new ArrayList<>(count);
+        for(int i = 0; i < count; i++) {
```

Comment:
@dimitris-athanasiou but that would then include all results. I calculate `count` so that we only create as many entries and add them to the list as necessary.

Reply:
Yeah, I was testing you 🙈 

(walks away whistling a fine tune)